File: frontend/src/api/documentsAdmin.ts
import api from "./client";

/** 집계 기준 — 일별 / 사용자별 / 종류별. */
export type UsageGroupBy = "day" | "user" | "kind";

/** kind 필터 — 양식 작성(fill) / 문서 생성(generate) / 전체. */
export type UsageKindFilter = "fill" | "generate" | "all";

export interface UsageRow {
  bucket: string;
  kind: string | null;
  job_count: number;
  tokens_in: number;
  tokens_out: number;
  cost_usd: number;
}

export interface UsageTotals {
  job_count: number;
  tokens_in: number;
  tokens_out: number;
  cost_usd: number;
}

export interface UsageResponse {
  group_by: UsageGroupBy;
  start: string;
  end: string;
  rows: UsageRow[];
  totals: UsageTotals;
}

export interface UsageQuery {
  start?: string; // YYYY-MM-DD
  end?: string; // YYYY-MM-DD
  group_by?: UsageGroupBy;
  kind?: UsageKindFilter;
}

/** 관리자 전용 문서 토큰/비용 사용량 집계. require_admin(403). */
export async function getDocumentsUsage(
  q: UsageQuery = {},
): Promise<UsageResponse> {
  const params: Record<string, string> = {};
  if (q.start) params.start = q.start;
  if (q.end) params.end = q.end;
  if (q.group_by) params.group_by = q.group_by;
  if (q.kind) params.kind = q.kind;
  const res = await api.get<UsageResponse>("/api/documents/admin/usage", {
    params,
  });
  return res.data;
}

File: backend/app/schemas/documents_admin.py
"""관리자 전용 문서 사용량(토큰/비용) 집계 스키마 (PR-E #1).

form_jobs 단일 테이블을 kind(fill/generate) 일반화 후 집계해 관리자 패널에 노출한다.
"""
from __future__ import annotations

from datetime import date
from typing import Literal

from pydantic import BaseModel

GroupBy = Literal["day", "user", "kind"]
KindFilter = Literal["fill", "generate", "all"]


class UsageRow(BaseModel):
    """집계 1행 — group_by 기준 버킷 + 합계."""

    bucket: str          # day: 'YYYY-MM-DD', user: 표시명, kind: 'fill'/'generate'
    kind: str | None     # group_by != 'kind' 면 None(여러 종류 합산 가능)
    job_count: int
    tokens_in: int
    tokens_out: int
    cost_usd: float


class UsageTotals(BaseModel):
    """기간 전체 합계."""

    job_count: int
    tokens_in: int
    tokens_out: int
    cost_usd: float


class UsageResponse(BaseModel):
    group_by: GroupBy
    start: date
    end: date
    rows: list[UsageRow]
    totals: UsageTotals
